src/main/java/uk/gov/crowncommercial/dts/scale/cat/model/entity/ProcurementEvent.java

```diff
@@ -45,4 +53,29 @@
   @Column(name = "updated_at")
   Instant updatedAt;
 
+  /**
+   * Builds an instance from basic details
+   *
+   * @param jaggaerEventId The rfx reference code
+   * @param principal
+   * @return a procurement event
+   */
+  public static ProcurementEvent of(Integer projectId, String eventName, String jaggaerEventId,
+      String ocdsAuthority, String ocidPrefix, String principal) {
+    var procurementEvent = new ProcurementEvent();
+    procurementEvent.setProjectId(projectId);
```

Comment:
Would it be preferable to add the new `ProcurementEvent` to the `ProcurementProject` entity's collection of events, and have the persistence layer then handle setting the foreign key column value on a save to the ProcurementProject repo..? Dunno - just thinking I can't remember explicitly setting the FK value before..

Reply:
What about if it was a proper bidirectional relationship? (which I think is what my original intention was - this probably was a first cut which just got left there - agree setting the foreignKey value explicitly isn't great - missed that)

Reply:
Yep absolutely - or that, probably more use long term 👍